osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/config/messaging/DefaultJmsConfiguration.java

```diff
@@ -0,0 +1,258 @@
+
+    @Value("${jms.default.connection.pool.block.if.session.pool.is.full.timeout:-1}")
+    private long jmsDefaultConnectionPoolBlockIfSessionPoolIsFullTimeout;
+
+    @Value("${jms.default.connection.pool.expiry.timeout:0}")
+    private long jmsDefaultConnectionPoolExpiryTimeout;
+
+    @Value("${jms.default.connection.pool.time.between.expiration.check.millis:-1}")
+    private long jmsDefaultConnectionPoolTimeBetweenExpirationCheckMillis;
+
+    @Value("${jms.default.connection.pool.idle.timeout:30000}")
+    private int jmsDefaultConnectionPoolIdleTimeout;
+
+    @Value("${jms.default.connection.queue.prefetch:1000}")
+    private int jmsDefaultConnectionQueuePrefetch;
+
+    @Value("${jms.default.connection.send.timeout:0}")
+    private int jmsDefaultConnectionSendTimeout;
+
+    @Value("${jms.default.trust.all.packages:true}")
+    private boolean jmsDefaultTrustAllPackages;
+
+    @Value("${jms.default.trusted.packages:org.opensmartgridplatform,org.joda.time,java.util}")
+    private String jmsDefaultTrustedPackages;
+
+    @Value("${jms.default.broker.client.key.store:/etc/osp/activemq/client.ks}")
+    private String jmsDefaultBrokerClientKeyStore;
+
+    @Value("${jms.default.broker.client.key.store.pwd:password}")
+    private String jmsDefaultBrokerClientKeyStorePwd;
+
+    @Value("${jms.default.broker.client.trust.store:/etc/osp/activemq/client.ts}")
+    private String jmsDefaultBrokerClientTrustStore;
+
+    @Value("${jms.default.broker.client.trust.store.pwd:password}")
+    private String jmsDefaultBrokerClientTrustStorePwd;
+
+    @Value("${jms.default.broker.username:#{null}}")
+    private String jmsDefaultBrokerUsername;
+
+    @Value("${jms.default.broker.password:#{null}}")
+    private String jmsDefaultBrokerPassword;
+
+    // Message Listener Container fields
+    @Value("${jms.default.max.concurrent.consumers:10}")
+    private int jmsDefaultMaxConcurrentConsumers;
+
+    @Value("${jms.default.concurrent.consumers:2}")
+    private int jmsDefaultConcurrentConsumers;
+
+    @Value("${jms.default.use.exponential.back.off:false}")
+    private boolean jmsDefaultUseExponentialBackOff;
+
+    @Value("${jms.default.back.off.multiplier:5}")
+    private double jmsDefaultBackOffMultiplier;
+
+    @Value("${jms.default.maximum.redelivery.delay:-1}")
+    private long jmsDefaultMaximumRedeliveryDelay;
+
+    @Value("${jms.default.initial.redelivery.delay:1000}")
+    private long jmsDefaultInitialRedeliveryDelay;
+
+    @Value("${jms.default.maximum.redeliveries:6}")
+    private int jmsDefaultMaximumRedeliveries;
+
+    @Value("${jms.default.redelivery.delay:1000}")
+    private long jmsDefaultRedeliveryDelay;
+
+    @Value("${jms.default.delivery.persistent:true}")
+    private boolean jmsDefaultDeliveryPersistent;
+
+    @Value("${jms.default.time.to.live:3600}")
+    private long jmsDefaultTimeToLive;
+
+    @Value("${jms.default.explicit.qos.enabled:true}")
+    private boolean jmsDefaultExplicitQosEnabled;
```

Comment:
Using @Value has the future risk of scattered configuration throughout the application. To avoid this, model configuration as a service. This means there should be an @Service or @Component somewhere which only concerns itself with providing access to the configuration values. Is this the case already?

Note: Spring boot offers @ConfigurationProperties to easily realize such a service (see https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/context/properties/ConfigurationProperties.html)
Spring

Reply:
In the current implementation the default configurations are loaded as beans from within @Configuration classes in the application.config package.
Unfortunately the OSGP components are not (yet) using Spring Boot.